include/modem/location.h

```diff
@@ -0,0 +1,343 @@
+	 * 'methods_count'.
+	 */
+	struct location_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
+	 */
+	uint16_t interval;
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct location_event_data *event_data);
+
+/**
+ * @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ * This can be called multiple times, which sets the event handler again.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EINVAL Given event_handler is NULL.
+ * @retval -EFAULT Failed to obtain Wi-Fi interface.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ * configuration, a single position or periodic position updates are given. If the position
+ * request is successful, the results are delivered to the event handler function in
+ * LOCATION_EVT_LOCATION event. Otherwise LOCATION_EVT_TIMEOUT or LOCATION_EVT_ERROR event is
+ * triggered.
+ *
+ * In periodic mode, position update is always attempted after the configured interval
+ * regardless of the outcome of any previous attempt. Periodic position updates can be
+ * stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EPERM  Library not initialized.
+ * @retval -EBUSY  Previous location request still ongoing.
+ * @retval -EINVAL Invalid configuration given.
+ */
+int location_request(const struct location_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EPERM  Library not initialized.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets the default values to a given configuration.
+ *
+ * @details Given methods are set as a list of methods for the given configuration.
+ * Caller is responsible for allocating and deallocating memory for both configuration and
+ * method list. The 'methods' memory area must be kept allocated as long as the configuration
+ * is used. Using a statically allocated list is encouraged.
+ *
+ * @param[in,out] config Configuration to be supplied with default values.
+ * @param[in] methods_count Number of methods to be stored in 'methods'.
+ * @param[in] methods List of methods. This will be initialized to zero.
+ */
+void location_config_defaults_set(
+	struct location_config *config,
```

Comment:
`location_config` contains a `methods_count`; does this need to match the methods count specified as the second parameter to this function?

Reply:
This function will set `methods_count` parameter into the `methods_count` of given `location_config`.
This function has also been changed today. Assumption is that when calling this function, given `location_config` has random values and this function will set defaults overriding anything that's been set earlier.